lib/interface_quda.cpp

```diff
@@ -9178,116 +9103,55 @@ void calcMG_loop_wOneD_TSM_wExact(void **gaugeToPlaquette,
-	dirac.reconstruct(*x_LP,*b,param->solution_type);
-	
-	// Revert to the original, high-precision values
-	if(TSM_maxiter==0) param->tol = orig_tol;           
-	else if(TSM_tol==0) param->maxiter = orig_maxiter;      
-	//-------------------------------------------------
-
-      }
-      else{
-	if(stochEO) {
-	  //Should always be false, use MG rather than
-	  //delflating the stochastic remainder.
-	  deflationEO->deflateVector(*K_vecdef,*K_vector);
-	}
-	else {
-	  //Calculation should default to this.
-	  //Project out the exact part only.
-	  deflation->deflateVector(*K_vecdef,*K_vector);
-	}
-	t2 = MPI_Wtime();
-	printfQuda("TIME_REPORT: NHP %04d - Init.guess deflation: %f sec\n",
-		   is+1,t2-t1);
-	// Initial guess is deflated to: out = U(\Lambda^-1)U^dag b
-	K_vecdef->uploadToCuda(out   ,flag_eo);
-	K_vecdef->uploadToCuda(out_LP,flag_eo);
-	
-	//HP solve
-	//-------------------------------------------------
-	SolverParam solverParam(*param);
-	Solver *solve = Solver::create(solverParam, m, mSloppy, mPre, 
-				       profileInvert);
-	(*solve)   (*out,*in);
-	delete solve;
-	dirac.reconstruct(*x,*b,param->solution_type);
-	//-------------------------------------------------
-	
-	
-	//LP solve
-	//-------------------------------------------------
-	double orig_tol = param->tol;
-	long int orig_maxiter = param->maxiter;
-	// Set the low-precision criterion
-	if(TSM_maxiter==0) param->tol = TSM_tol;
-	else if(TSM_tol==0) param->maxiter = TSM_maxiter;  
-	
-	// Create the low-precision solver
-	SolverParam solverParam_LP(*param);
-	Solver *solve_LP = Solver::create(solverParam_LP, m, mSloppy, 
-					  mPre, profileInvert);
-	(*solve_LP)(*out,*in);
-	delete solve_LP;
-	dirac.reconstruct(*x_LP,*b,param->solution_type);
-	
-	// Revert to the original, high-precision values
-	if(TSM_maxiter==0) param->tol = orig_tol;           
-	else if(TSM_tol==0) param->maxiter = orig_maxiter;      
-	//-------------------------------------------------
-	
-	sol    = new cudaColorSpinorField(*x);
-	sol_LP = new cudaColorSpinorField(*x_LP);
-      }
-
+      //HP solve
+      //-------------------------------------------------
+      SolverParam solverParam(*param);
+      Solver *solve = Solver::create(solverParam, m, mSloppy, mPre, 
+				     profileInvert);
+      (*solve)   (*out,*in);
+      delete solve;
+      dirac.reconstruct(*x,*b,param->solution_type);
+      //-------------------------------------------------
+      
+      
+      //LP solve
+      //-------------------------------------------------
+      double orig_tol = param->tol;
+      long int orig_maxiter = param->maxiter;
+      // Set the low-precision criterion
+      if(TSM_maxiter==0) param->tol = TSM_tol;
+      else if(TSM_tol==0) param->maxiter = TSM_maxiter;  
+      
+      // Create the low-precision solver
+      SolverParam solverParam_LP(*param);
+      Solver *solve_LP = Solver::create(solverParam_LP, m, mSloppy, 
+					mPre, profileInvert);
+      (*solve_LP)(*out,*in);
```

Comment:
line 9131: It should be: (*solve_LP)(*out_LP,*in);
I.e. replace "out" with "out_LP"